Fixed node v6 deprecation message, removed lodash, added only updated lodash isString and isFunction

hinderberg opened the pull request.

davidmurdoch commented on easy-pbkdf2.js:
```diff
@@ -1,15 +1,15 @@
-		_.each( options, function( value, key ){
-			this[ key ] = value;
-		}, this);
+	if (options instanceof Object) {
```
`options instanceof Object` is a poor alternative to isPlainObject
because it will return true for functions, arrays, and any other non-plain object.

`[] instanceof Object // true`

Can you update this line to `options && typeof options === "object"`?

davidmurdoch commented on easy-pbkdf2.js:
```diff
@@ -203,7 +203,7 @@ EasyPbkdf2.prototype = {
 			easyPbkdf2;
 
-		if ( !priorHash || !_.isString( priorHash ) ) {
+		if ( !priorHash || !( typeof priorHash === "string" ) ) {
```
Unfortunately, this check would now fail if `priorHash` was created via `new String()`... and I don't want to introduce a potentially breaking change here. I am tempted to say that anyone creating strings this way deserves to have their code broken, I just can't do it. I wish I had written it this way from the start.

I'm curious, why do you want to remove lodash as a dep?

davidmurdoch commented on easy-pbkdf2.js:
```diff
@@ -72,7 +72,7 @@ EasyPbkdf2.prototype = {
 	 */
 	"random": function( bytes, callback ) {
-		if ( _.isFunction( callback ) ) {
+		if ( callback instanceof Function ) {
```
I'd like to avoid using `instanceof` checks because they'll fail for values from other realms/contexts (like node's `vm`). No idea why someone would be passing in functions from other contexts in here, but it could happen.

hinderberg commented:
I just fixed some of your comments. The reason I removed lodash in the first place was because it was outdated (i.e deprecated) and in my opinion it would be enough to use vanilla JS. But as long as the deps are updated regularly, it's not a problem. The main fix here was to update the `crypto.pbkdf2` to not use a deprecated api. I also removed all the spaces between functions and if's since some places it was used, others not. Better to be consistent.

hinderberg commented:
@davidmurdoch any feedback?

davidmurdoch commented:
@hinderberg, sorry for the delay. That last commit changed so much I really didn't have time to look at it. If  you want to send it again without changing quotes or whitespace I'll try to get to it this week.

p.s., I generally try to follow [jQuery's spacing styleguide](https://contribute.jquery.org/style-guide/js/#spacing) (though not perfectly, as evident by my inconsistencies).

dawnerd commented:
Any chance this can be merged in? Would love to get the node 6 fix.

davidmurdoch commented:
@dawnerd, I'm just need the spacing put back the way it was and the quotes back to double quotes. There is too much noise in these commits to pull them in as is.
